lib/spack/spack/solver/asp.py

```diff
@@ -545,9 +563,13 @@ def solve(
         self.assumptions = []
         with self.control.backend() as backend:
             self.backend = backend
-            solver_setup.setup(self, specs, tests=tests, reuse=reuse)
+            setup.setup(self, specs)
         timer.phase("setup")
 
+        # If we're only doing setup, just return an empty solve result
+        if setup_only:
+            return Result(specs)
+
```

Comment:
I think this is wrongly positioned, we should exit _after_ we generated the error facts. I'll submit a bug fix asap.